contracts/relayer/RelayerAragonAppWithParameterizedSender.sol

```diff
@@ -0,0 +1,29 @@
+pragma solidity ^0.4.24;
+
+import "./BaseRelayer.sol";
+
+
+contract RelayerAragonAppWithParameterizedSender is BaseRelayer {
+    modifier relayedAuth(address _sender, bytes32 _role) {
+        require(canPerform(_sender, _role, new uint256[](0)), ERROR_AUTH_FAILED);
+        _;
+    }
+
+    modifier relayedAuthP(address _sender, bytes32 _role, uint256[] _params) {
+        require(canPerform(_sender, _role, _params), ERROR_AUTH_FAILED);
+        _;
+    }
+
+    function exec(address from, uint256 nonce, bytes calldata, bytes signature) external refundGas auth(OFF_CHAIN_RELAYER_SERVICE_ROLE) {
+        assertValidTransaction(from, nonce, calldata, signature);
```

Comment:
The issue that I see with this approach is that ensuring that the sender in the calldata is the signer of the message would require inspecting the calldata to extract the sender parameter and checking that it equals to `from`. 

With the current implementation, any address can sign a valid message that would allow them to impersonate someone else.

Reply:
Yea, being honest, this approach doesn't look good at all to me.